packages/lib/slots.ts

```diff
@@ -192,18 +195,24 @@ function buildSlotsWithDateRanges({
       ? range.start
       : startTimeWithMinNotice;
 
-    slotStartTime =
-      slotStartTime.minute() % interval !== 0
-        ? slotStartTime.startOf("hour").add(Math.ceil(slotStartTime.minute() / interval) * interval, "minute")
-        : slotStartTime;
-
     // Adding 1 minute to date ranges that end at midnight to ensure that the last slot is included
     const rangeEnd = range.end
       .add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes")
       .isSame(range.end.endOf("day").add(dayjs().tz(organizerTimeZone).utcOffset(), "minutes"), "minute")
       ? range.end.add(1, "minute")
       : range.end;
 
+    slotStartTime =
+      slotStartTime.minute() % interval !== 0
+        ? showOptimizedSlots
+          ? rangeEnd.diff(slotStartTime, "minutes") % interval > interval - slotStartTime.minute()
+            ? slotStartTime.add(interval - slotStartTime.minute(), "minute")
+            : slotStartTime.add(rangeEnd.diff(slotStartTime, "minutes") % interval, "minute")
+          : slotStartTime
+              .startOf("hour")
+              .add(Math.ceil(slotStartTime.minute() / interval) * interval, "minute")
+        : slotStartTime;
+
```

Comment:
Avoid ternaries within ternaries, it not not very readable and can quickly become a nightmare to debug. Can you please refactor this bit?

Reply:
sure will do

Reply:
@Amit91848 , have refactored and added comments, can you please see if it' ok now.